lib/sass/tree/visitors/perform.rb

```diff
@@ -346,29 +346,47 @@ def visit_mixindef(node)
 
   # Runs a mixin.
   def visit_mixin(node)
+    mixin_name = nil
     @environment.stack.with_mixin(node.filename, node.line, node.name) do
-      mixin = @environment.mixin(node.name)
-      raise Sass::SyntaxError.new("Undefined mixin '#{node.name}'.") unless mixin
-
-      if node.children.any? && !mixin.has_content
-        raise Sass::SyntaxError.new(%(Mixin "#{node.name}" does not accept a content block.))
-      end
-
       args = node.args.map {|a| a.perform(@environment)}
       keywords = Sass::Util.map_vals(node.keywords) {|v| v.perform(@environment)}
       splat = self.class.perform_splat(node.splat, keywords, node.kwarg_splat, @environment)
 
+      mixin_name = node.name
+      # We let any existing mixin named `mixin` shadow dynamic includes
+      # for backwards compatability even though it will break the hell
+      # out of libraries if anyone does this.
+      mixin = @environment.mixin(mixin_name)
+
+      # The while loop is to allow any number of dynamic includes to be chained.
+      while mixin.nil? && mixin_name == "mixin"
```

Comment:
> provide support for arbitrary Ruby-defined mixins

I thought about this. It's an experimental feature that libsass has, actually. I don't love the idea of people writing mixins in ruby, tbh. I think functions are cleaner for calling out to the host language. Even if it was just an internal API, my sense is that we should hold off on doing that until we have a second or third use case to fully understand the problem space.